src/LayerPlanBuffer.cpp

```diff
@@ -17,16 +17,55 @@ void LayerPlanBuffer::flush()
     }
     while (!buffer.empty())
     {
-        buffer.front().writeGCode(gcode);
+        buffer.front()->writeGCode(gcode);
         if (CommandSocket::isInstantiated())
         {
             CommandSocket::getInstance()->flushGcode();
         }
         buffer.pop_front();
     }
-    
 }
 
+void LayerPlanBuffer::addConnectingTravelMove()
+{
+    auto newest_layer_it = --buffer.end();
+    auto prev_layer_it = newest_layer_it;
+    prev_layer_it--;
+    Point destination;
+    {
+        const LayerPlan* newest_layer = *newest_layer_it;
+        assert(newest_layer->extruder_plans.front().extruder == (*prev_layer_it)->extruder_plans.back().extruder);
+        const ExtruderPlan& extruder_plan_above = newest_layer->extruder_plans.front();
+        if (extruder_plan_above.paths.size() == 0)
+        {
+            logWarning("There are empty layers (or layers with empty extruder plans) in the print! Temperature control and cross layer travel moves might suffer.\n");
+            return;
+        }
+        const GCodePath& path_above = extruder_plan_above.paths[0];
+        assert(path_above.points.size() == 1 && "The first move planned should have been a boguous direct travel move");
```

Comment:
I could try to leave out the travel move at the start of a layer, and instead have something like `extrusion_start_position` as a field in the layer plan.
However, this wouldn't work so well, because the bogus travel move is only inserted for the first extruder plan in a layer.
When switching extruder we first need to move to the switching bay, then start the new extruder plan with a combing travel move to the first extrusion position. If we add that combing move to the last extruder plan we would comb toward the object before we actually switch the extruder.

I do agree with your comment, but I don't really see a nice way to solve this. I'm not sure we should spend that much time on it, given that the 2.5 beta release is around the corner.